### PR TITLE
move server config to easy to extend style

### PR DESCRIPTION
**Pull request by tjdevries**
Alternative to #54 

@j2501987804 what do you think of this style instead?

### CODE REVIEW

**Comment by craigmac** [comment]
Easier to follow, I'd say, and easier to extract should you not want to set them up via mason lsp-config.

File: init.lua [hunk 1]
```diff
@@ -332,63 +332,57 @@ local on_attach = function(_, bufnr)
...
+  ensure_installed = vim.tbl_keys(servers),
 }
 
--- nvim-cmp supports additional completion capabilities
+-- nvim-cmp supports additional completion capabilities, so broadcast that to servers
 local capabilities = vim.lsp.protocol.make_client_capabilities()
 capabilities = require('cmp_nvim_lsp').default_capabilities(capabilities)
 
-for _, lsp in ipairs(servers) do
+for lsp, settings in ipairs(servers) do
```
Comment (MaxVerevkin) [inline init.lua:375]: This should be `pairs`, I think.

**Comment by tjdevries** [comment]
@craigmac @MaxVerevkin thoughts on new style? found a new setup that's pretty nice for mason-lspconfig that I wasn't aware of from #74

**Comment by MaxVerevkin** [comment]
I like it. It also allows setting up new servers without editing the config (i.e. I can install a server and it will just work).

**Comment by craigmac** [comment]
Ditto.

**Comment by zdzolton** [comment]
(Apologies for commenting on a closed PR; I will gladly move elsewhere!)

Does this change allow for users to extend the language server list without directly modifying the init.lua file? I've been looking at using Kickstart by symlinking into my config directory [as mentioned here](https://github.com/nvim-lua/kickstart.nvim/#configuration) but haven't found a way to do what I want... Any ideas?

**Comment by MaxVerevkin** [comment]
> extend the language server list without directly modifying the init.lua file?

Yes, just install them using `:Mason`.

**Comment by zdzolton** [comment]
>Yes, just install them using :Mason.

Got it... Thank you, @MaxVerevkin!

I was worried that servers not added to the the local `servers` variable would miss out the completion capabilities configured here: 
https://github.com/nvim-lua/kickstart.nvim/blob/521940693e10eee57f4e67dce960d2229c1441d3/init.lua#L357-L379

**Comment by tjdevries** [comment]
Nope, pretty sure they get picked up. But you should read Mason documentation to understand that better (and I should probably too haha)

**Comment by j2501987804** [comment]
image
It seems that not all languages are settings, such as gopls, which provides {cmd,filetypes,filetypes} attributes

**Pull request merged**